swarm/api/http/server_test.go

```diff
@@ -892,3 +894,268 @@ func TestMethodsNotAllowed(t *testing.T) {
+		}
+	}
+	putPatchHash()
+
+	// DELETE bzz:/hash -> 200 OK
+	deleteHash := func() {
+		url := fmt.Sprintf("%s/bzz:/%s", srv.URL, hash)
+
+		res, body := httpDo("DELETE", url, nil, nil, verbose, t)
+
+		if res.StatusCode != 200 {
+			t.Fatal("expected DELETE bzz:/hash to return 200 but it didn't")
+		}
+		if body != "8b634aea26eec353ac0ecbec20c94f44d6f8d11f38d4578a4c207a84c74ef731" {
+			t.Fatal("After DELETE bzz:/hash, the manifest's hash is not the value we expected")
+		}
+	}
+	deleteHash()
+
+	// POST bzz-raw:/ "POSTdata" -> 200 OK
+	postData := func() {
+		url := fmt.Sprintf("%s/bzz-raw:/", srv.URL)
+		buf := bytes.NewReader([]byte("POSTdata"))
+
+		res, hash := httpDo("POST", url, buf, nil, verbose, t)
+
+		if res.StatusCode != 200 {
+			t.Fatal("expected POST bzz-raw:/ to return 200 but it didn't")
+		}
+		// Try downloading what we just uploaded
+		url = fmt.Sprintf("%s/bzz-raw:/%s", srv.URL, hash)
+
+		res, body := httpDo("GET", url, nil, nil, verbose, t)
+		if body != "POSTdata" {
+			t.Fatalf("expected %s to be 'POSTdata' but it wasn't", hash)
+		}
+	}
+	postData()
+
+	// PUT bzz-raw:/hash -> 405 Method Not Allowed
+	putHash := func() {
+		url := fmt.Sprintf("%s/bzz-raw:/%s", srv.URL, hash)
+
+		res, _ := httpDo("PUT", url, nil, nil, verbose, t)
+
+		if res.StatusCode != 405 {
+			t.Fatal("expected PUT bzz-raw:/hash to return 405 but it didn't")
+		}
+	}
+	putHash()
+
+	// POST bzz-raw:/encrypt
+	postBzzRawEncrypt := func() {
+		url := fmt.Sprintf("%s/bzz-raw:/encrypt", srv.URL)
+		buf := bytes.NewReader([]byte("POSTdata"))
+
+		res, body := httpDo("POST", url, buf, nil, verbose, t)
+
+		if res.StatusCode != 200 {
+			t.Fatalf("expected POST bzz-raw:/encrypt to return 200, but got %d", res.StatusCode)
+		}
+		if len(body) != 128 {
+			t.Fatalf("encrypted uploads via POST bzz-raw:/encrypt should return a 128 char long reference, but this was only %d", len(body))
+		}
+	}
+	postBzzRawEncrypt()
+
+	// POST /bzz:/ Content-Type: multipart/form-data
+	postBzzMultipartFormData := func() {
+		url := fmt.Sprintf("%s/bzz:/", srv.URL)
+
+		buf := new(bytes.Buffer)
+		form := multipart.NewWriter(buf)
+		form.WriteField("name", "John Doe")
+		file1, _ := form.CreateFormFile("cv", "cv.txt")
+		file1.Write([]byte("John Doe's Credentials"))
+		file2, _ := form.CreateFormFile("profile_picture", "profile.jpg")
+		file2.Write([]byte("imaginethisisjpegdata"))
+		form.Close()
+
+		headers := make(map[string]string)
+		headers["Content-Type"] = form.FormDataContentType()
+		headers["Content-Length"] = strconv.Itoa(buf.Len())
+		res, body := httpDo("POST", url, buf, headers, verbose, t)
+
+		if res.StatusCode != 200 {
+			t.Fatalf("expected POST multipart/form-data to return 200, but it returned %d", res.StatusCode)
+		}
+		if len(body) != 64 {
+			t.Fatalf("expected POST multipart/form-data to return a 64 char manifest but the answer was %d chars long", len(body))
+		}
+	}
+	postBzzMultipartFormData()
```

Comment:
remove this line and on the line above just `}()` if you really insist on having this in a closure.
although i'd be happy to see this move into a separate `func TestMultipartUpload(t *testing.T)`